src/Controllers/controller.class.php

```diff
@@ -81,7 +83,19 @@ public function call(string $method, ?array $args = []) : mixed {
         if (!method_exists($this, $method)) {
             throw new MethodNotFoundException('La méthode ' . $method . ' n\'existe pas dans le contrôleur ' . get_class($this));
         }
-        return $this->{$method}(...array_values($args));
+        try {
+            return $this->{$method}(...array_values($args));
+        } catch (Exception $e) {
+            switch ($e->getCode()) {
+                case 401:
+                    ErrorHandler::addExceptionParametersToSession($e);
+                    header('Location: /login');
+                    break;
+                default:
+                    ErrorHandler::displayFullScreenException($e);
+                    break;
+            }
```

Comment:
La logique avec le `$e->getCode()` n'est pas bonne car pas assez précise.
Utiliser plutôt les nom des exceptions :
```suggestion
            switch ($e::class) {
                case AuthenticationException::class:
                    ErrorHandler::addExceptionParametersToSession($e);
                    header('Location: /login');
                    break;
                default:
                    ErrorHandler::displayFullScreenException($e);
                    break;
            }
```

Reply:
Il faut que je vérif le reste du traitement de l'exception.
Mais je pense que le mieux est que tu mettes un @todo pour y penser plus tard.